### ISSUE TITLE
NoSuchMethodError with Jack compiler

### ISSUE THREAD

**Issue by elevenetc** [google/dagger#557]
Getting next error with Jack compiler:
```
Caused by: java.lang.NoSuchMethodError: com.google.common.collect.Sets$SetView.iterator()Lcom/google/common/collect/UnmodifiableIterator;
        at dagger.internal.codegen.ModuleProcessingStep.process(ModuleProcessingStep.java:147)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:329)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:182)
        at com.android.jack.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:139)
        ... 94 more
```
To reproduce problem here is sample project: https://github.com/elevenetc/dagger-and-jack-sample-exception
To build sample:
```
./gradlew clean assembleDebug --stacktrace 
```
If jack is disabled
```
jackOptions {
    enabled false
}
```
then app builds successfully.

**Comment by ronshapiro** [comment]
Sounds related to #507

**Comment by ronshapiro** [comment]
I've filed https://code.google.com/p/android/issues/detail?id=232384 for this since it continues to come up.

**Comment by elevenetc** [comment]
Temporary solution is to use `2.3.0-beta2` plugin with 3.3 Gradle:
> This should be fixed in 2.3.0-beta2. Could you please try to upgrade the plugin (you will have to upgrade Gradle wrapper to 3.3)?
